File: CS/DXSample/Data/SimpleDataBase.cs
using System;
using System.Linq;
using System.ComponentModel;
using System.Linq.Expressions;
using System.Collections.Generic;

namespace DXSample {
    public class SimpleDataBase : INotifyPropertyChanged {

        public event PropertyChangedEventHandler PropertyChanged;

        protected static string GetPropertyName<T>(Expression<Func<T>> expression) {
            string propertyName;
            var memberExpression = expression as MemberExpression;
            if (memberExpression != null)
                propertyName = memberExpression.Member.Name;
            else
                propertyName = string.Empty;
            return propertyName;
        }

        protected void OnPropertyChanged(string property) {
            if (PropertyChanged != null) {
                PropertyChanged(this, new PropertyChangedEventArgs(property));
            }
        }
    }
}

File: VB/DXSample/Data/SimpleDataBase.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.ComponentModel
Imports System.Linq.Expressions
Imports System.Collections.Generic

Namespace DXSample
	Public Class SimpleDataBase
		Implements INotifyPropertyChanged

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        

		Protected Sub OnPropertyChanged(ByVal [property] As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs([property]))
		End Sub
	End Class
End Namespace
